csrc/layernorm_quant_kernels.cu

```diff
@@ -20,40 +20,67 @@
 
 namespace vllm {
 
-// TODO(woosuk): Further optimize this kernel.
+template <typename T>
+__device__ __forceinline__ T warp_sum(T v) {
+#ifdef __HIP_PLATFORM_AMD__
+  const unsigned long long m = 0xffffffffffffffffull;  // HIP needs 64-bit mask
+#else
+  const unsigned m = 0xffffffffu;
+#endif
+  constexpr int kWidth = 32;  // keep reduction over 32 lanes everywhere
+  v += __shfl_down_sync(m, v, 16, kWidth);
+  v += __shfl_down_sync(m, v, 8, kWidth);
+  v += __shfl_down_sync(m, v, 4, kWidth);
+  v += __shfl_down_sync(m, v, 2, kWidth);
+  v += __shfl_down_sync(m, v, 1, kWidth);
+  return v;
+}
```

Comment:
Seems identical with previous code, is there any way we can reuse it instead of copying it twice?